org.testeditor.web.dropwizard.xtext/src/test/java/org/testeditor/web/xtext/index/BuildCycleManagerIntegrationTest.xtend

```diff
@@ -0,0 +1,196 @@
+		val root = new File(tmpDir.root, remoteRoot)
+		val remoteGit = Git.init.setDirectory(root).call
+		write(root, 'build.gradle', '''apply plugin: 'java' ''')
+		addAndCommit(remoteGit, 'build.gradle', 'Add build.gradle')
+		write(root, 'gradlew', '''
+			#!/bin/bash
+			echo "running dummy gradle ..."
+			if [ "$1" == "tasks" ]; then
+			  echo "printTestClasspath"
+			elif [ "$1" == "printTestClasspath" ]; then
+			  echo "«root.absolutePath»/mydsl.jar"
+			fi
+		''')
+		new File(root, 'gradlew').executable = true
+		addAndCommit(remoteGit, 'gradlew', 'Add dummy gradlew')
+		write(root, 'src/test/java/Demo.mydsl', 'Hello Peter!')
+		addAndCommit(remoteGit, 'src/test/java/Demo.mydsl', 'Add MyDsl.xtext as an example')
+
+		gitService.init(config.localRepoFileRoot, config.remoteRepoUrl)
+	}
+
+	@Test
+	def void detectChangesReturnsModifiedFiles() {
+		// given
+		val initialBuildRequest = new BuildRequest
+
+		// when
+		val actualBuildRequest = initialBuildRequest.addChanges
+
+		// then
+		assertThat(actualBuildRequest.dirtyFiles).containsOnly(URI.createFileURI(config.localRepoFileRoot + '/src/test/java/Demo.mydsl'))
+	}
+
+	@Test
+	def void createBuildRequestSetsRequiredFields() {
+		// when
+		val actualBuildRequest = unitUnderTest.createBuildRequest
+
+		// then
+		assertThat(actualBuildRequest.baseDir).isEqualTo(URI.createFileURI(config.localRepoFileRoot))
+		assertThat(actualBuildRequest.resourceSet).isInstanceOf(XtextResourceSet)
+		assertThat(actualBuildRequest.afterValidate).isEqualTo(validationMarkerUpdater)
+		assertThat(actualBuildRequest.state.getResourceDescriptions.exportedObjects).isEmpty
+	}
+
+	@Test
+	def void createBuildRequestAlwaysUsesSameResourceSet() {
+		// given
+		val firstBuildRequest = unitUnderTest.createBuildRequest
+
+		// when
+		val secondBuildRequest = unitUnderTest.createBuildRequest
+
+		// then
+		assertThat(firstBuildRequest.resourceSet).isSameAs(secondBuildRequest.resourceSet)
+	}
+
+	@Test
+	def void launchReturnsUpdatedIndexState() {
+		// given
+		val buildRequest = sampleBuildRequest
+
+		// when
+		val actualIndexState = unitUnderTest.build(buildRequest)
+
+		// then
+		assertThat(actualIndexState.resourceDescriptions.exportedObjects.head.qualifiedName.toString).isEqualTo('Peter')
+	}
+
+	@Test
+	def void launchInvokesValidationMarkerUpdaterState() {
+		// given
+		val buildRequest = sampleBuildRequest
+
+		// when
+		unitUnderTest.build(buildRequest)
+
+		// then
+		verify(validationMarkerUpdater).afterValidate(any, any)
+	}
+
+	@Test
+	def void updateIndexPublishesNewIndexState() {
+		// given
+		val exportedObjectNames = #['modelElement', 'anotherElement']
+		val newIndexState = getMockedIndexState(exportedObjectNames)
+		val indexResourceSet = unitUnderTest.createBuildRequest.resourceSet
+
+		// when
+		unitUnderTest.updateIndex(newIndexState)
+
+		// then
+		assertThat(indexProvider.getResourceDescriptions(indexResourceSet).exportedObjects.head.qualifiedName.toString).isEqualTo('modelElement')
```

Comment:
anotherElement is not checked, it is not checked that no other exportedObject is present?